draft-ietf-tls-esni.md

```diff
@@ -347,16 +351,53 @@ For servers (in EncryptedExtensions), this extension contains the following
 structure:
 
 ~~~
+   enum {
+       esni_accept(0),
+       esni_retry_request(1),
+   } ServerESNIResponseType;
+
    struct {
-       uint8 nonce[16];
+       ServerESNIResponseType response_type;
+       select (response_type) {
+           case esni_accept:        uint8 nonce[16];
+           case esni_retry_request: ESNIKeys retry_keys<1..2^16-1>;
+       }
    } ServerEncryptedSNI;
 ~~~
 
+response_type
+: Indicates whether the server processed the client ESNI extension. (See
+{{handle-server-response}} and {{server-behavior}}.}
+
 nonce
 : The contents of ClientESNIInner.nonce. (See {{client-behavior}}.)
 
+retry_keys
+: ESNIKeys structures containing the keys that the client should use on
+subsequent connections to encrypt the ClientESNIInner structure.
+
+This protocol also defines the "esni_required" alert, which is sent by the
+client when it offered an "encrypted_server_name" extension which was not
+accepted by the server.
+
+~~~~
+   enum {
+       esni_required(121),
+   } AlertDescription;
+~~~~
+
+Finally, requirements in {{client-behavior}} and {{server-behavior}} require
+implementations to track, alongside each PSK established by a previous
+connection, whether the connection which this extension with the "esni_accept"
```

Comment:
I think this sentence is missing something...

Reply:
Oops! The mistake was s/which/negotiated/.